x-pack/legacy/plugins/reporting/types.d.ts

```diff
@@ -175,13 +178,17 @@ export interface ESQueueWorker {
   on: (event: string, handler: any) => void;
 }
 
-export type ESQueueWorkerExecuteFn = (job: JobDoc, cancellationToken: any) => void;
+export type ESQueueCreateJobFn = (
+  jobParams: JobParams,
+  headers: ConditionalHeaders,
+  request: Request
+) => Promise<JobParams>;
 
-export interface ExportType {
-  jobType: string;
-  createJobFactory: any;
-  executeJobFactory: (server: KbnServer) => ESQueueWorkerExecuteFn;
-}
+export type ESQueueWorkerExecuteFn = (job: JobDoc, cancellationToken: any) => void;
+export type ImmediateExecuteFn = (
+  jobDocPayload: JobDocPayload,
+  request: Request
+) => Promise<JobDocOutputExecuted>;
```

Comment:
`JobDocPayload` must be specific to `csv_from_savedobject`, since it works with `ImmediateExecuteFn` which is basically CSV from Saved Object.